src/main/kotlin/org/rust/lang/core/resolve/ref/RsPathReferenceImpl.kt

```diff
@@ -152,35 +168,102 @@ class RsPathReferenceImpl(
         return element.replace(elementNew)
     }
 
-    private object Resolver : (RsPath) -> List<BoundElementWithVisibility<RsElement>> {
-        override fun invoke(element: RsPath): List<BoundElementWithVisibility<RsElement>> {
-            return resolvePath(element)
+    private object Resolver : (RsElement) -> Any {
+        /**
+         * Returns `List<RsPathResolveResult<RsElement>>` if [root] is a single path or
+         * `Map<RsPath, List<RsPathResolveResult<RsElement>>>` if [root] contains nested paths.
+         */
+        override fun invoke(root: RsElement): Any {
+            val allPaths = root.collectNestedPathsFromRoot(root)
+            if (allPaths.isEmpty()) return emptyList<RsPathResolveResult<RsElement>>()
+            val ctx = PathResolutionContext(allPaths.first(), isCompletion = false, null)
```

Comment:
```suggestion
            val ctx = PathResolutionContext(root, isCompletion = false, null)
```

Reply:
In the future I'm going to resolve even more paths at once. First is an `impl` signature - I want to resolve type and implemented trait at once. In this case, `root` will be not a path, but an impl item itself. But we can't use impl item as a context here because it's then used as a `scopeStart` in `processNestedScopesUpwards` and then we're skipping the `impl` scope. I can use `root` instead of `allPaths.first()` in this PR